tst/regression/test_suites/calculate_pi/calculate_pi.py

```diff
@@ -0,0 +1,94 @@
+#========================================================================================
+# Athena++ astrophysical MHD code
+# Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+# Licensed under the 3-clause BSD License, see LICENSE file for details
+#========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#========================================================================================
+
+# Modules
+import math
+import numpy as np
+import sys
+import os
+import utils.test_case
+
+""" To prevent littering up imported folders with .pyc files or __pycache_ folder"""
+sys.dont_write_bytecode = True
+
+class TestCase(utils.test_case.TestCaseAbs):
+    def Prepare(self,parameters):
+        """
+        Any preprocessing that is needed before the drive is run can be done in
+        this method
+
+        This includes preparing files or any other pre processing steps that
+        need to be implemented.  The method also provides access to the
+        parameters object which controls which parameters are being used to run
+        the driver. 
+
+        It is possible to append arguments to the driver_cmd_line_args if it is
+        desired to  override the parthenon input file. Each element in the list
+        is simply a string of the form '<block>/<field>=<value>', where the
+        contents of the string are exactly what one would type on the command
+        line run running a parthenon driver.
+
+        As an example if the following block was uncommented it would overwrite
+        any of the parameters that were specified in the parthenon input file
+        parameters.driver_cmd_line_args = ['output1/file_type=vtk',
+                'output1/variable=cons',
+                'output1/dt=0.4',
+                'time/tlim=0.4',
+                'mesh/nx1=400']
+        """
+        return parameters
+
+    def Analyse(self,parameters):
```

Comment:
One use case which I think this supports which might be useful is generating plots in `Analyse`. There are certain tests where a plot can tell us what happened if the test failed.

I don't think you need to explicitly support this in the infrastructure. The test writer can just add some code to generate a plot and save it if they think that's necessary.